web-app/web-frontend/js/chess.js

```diff
@@ -732,6 +730,49 @@ var Chess = function (fen) {
   
       return output
     }
+     
+    /* convert a move from 0x88 coordinates to Grid Algebraic Notation
+     * (GAN)
+     *
+     * displays move in the form of p1p2, p1 as the starting position and p2 as the destination position
+     * displays promotion move in the form of p1p2<lowercase-piece-type>
+     * displays p1p2+ if move creates a check 
+     * displays p1p2# if move creates a checkmate 
+     * 
+     */
+    function move_to_gan(move, moves) {
+      var output = ''
+      if (move.flags & BITS.KSIDE_CASTLE) {
+        output = 'O-O'
+      } else if (move.flags & BITS.QSIDE_CASTLE) {
+        output = 'O-O-O'
+      } else {
+        if (move.piece !== PAWN) {
+          var disambiguator = get_disambiguator(move, moves)
+          output += disambiguator
+        }
```

Comment:
What does this code do/is it necessary? I think you may be able to remove it for simplicity's sake.